.github/workflows/sync.yml

```diff
@@ -0,0 +1,76 @@
+name: Sync scriptlets
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 20 * * 1,5'
+
+jobs:
+  sync:
+    name: Update scriptlets
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    if: github.repository_owner == 'ghostery'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: jdx/mise-action@v2
+
+      - name: Sync with upstream
+        run: |
+          function gh_repo_tags() {
+            # REPO_REF="owner/repo"
+            local REPO_REF="$1"
+            # https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-tags
+            curl -sL \
+              -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "https://api.github.com/repos/$REPO_REF/tags"
+          }
+
+          function gh_repo_latest_tagname() {
+            # REPO_REF="owner/repo"
+            local REPO_REF="$1"
+            gh_repo_tags "$REPO_REF" | \
+            jq -r '.[].name | select(test("^[0-9]"))' | \
+            sort -Vr | \
+            head -1
+          }
+
+          function get_local_tagname() {
+            # deno build.ts --tagName 1.64.0 > ubo.js
+            cat 'package.json' | sed -n -E 's/.*--tagName ([0-9\.a-zA-Z]+).*/\1/p'
+          }
+
+          LOCAL_REF="$(get_local_tagname)"
+          UPSTREAM_REF="$(gh_repo_latest_tagname 'gorhill/ublock')"
+          if [[ "$UPSTREAM_REF" == 'null' ]]; then
+            echo "ERROR: Failed to retrieve upstream version! Run with 'set -xe' for debugging."
+            exit 1
+          fi
+          if [[ "$(echo "$LOCAL_REF\n$UPSTREAM_REF" | sort -Vr | head -1)" != "$LOCAL_REF" ]]; then
+            NEW_TAGNAME="$1"
+            cat 'package.json' | sed -E "s/(--tagName )([0-9\.a-zA-Z]+)/\1$NEW_TAGNAME/" | tee 'package.json'
+            deno run build
+          fi
+
```

Comment:
```suggestion
          #!/usr/bin/env bash
          set -euo pipefail

          # Fetch latest upstream tag from GitHub Releases
          upstream_tag=$(curl -s https://api.github.com/repos/gorhill/ublock/releases/latest \
              | jq -r .tag_name)

          if [[ "$upstream_tag" == null ]]; then
            echo "ERROR: could not fetch upstream tag"
            exit 1
          fi

          # Extract local tag from package.json
          local_tag=$(sed -nEn 's/.*--tagName ([0-9.]+).*/\1/p' package.json)

          # If upstream tag differs from local, update and rebuild
          if [[ "$upstream_tag" != "$local_tag" ]]; then
            sed -i -E "s/--tagName [0-9.]+/--tagName $upstream_tag/" package.json
            deno run build
          fi
```

Reply:
refactored in 01df394aac367c05b60069da580a4f5771c5286a but leaving some notes

- Kept regex patterns as new regexp don't support all cases e.g. `1.64.1b0`
- Kept the form of `cat ... | sed ... | tee ...` instead of sed in-place modification for straightforward compatibility with bsd sed (for direct macOS testing without any confusion)
- Sorting and filtering over tag name from gh response is still a thing as they respond with `npm` prefix which we don't want to use
- Kept github api version in curl for future consistency